packages/compass-query-bar/src/stores/query-bar-reducer.ts

```diff
@@ -57,6 +51,43 @@ export function isQueryFieldsValid(fields: Record<string, QueryBarFormField>) {
   );
 }
 
+function wrapFilterIfNotWrapped(
+  filter: QueryBarState['fields']['filter']
+): QueryBarState['fields']['filter'] {
+  if (validateField('filter', filter.string) !== false || !filter) {
+    return filter;
+  }
+
+  // When the filter isn't valid we try to wrap it in {}.
+  const filterWithBrackets = validateField('filter', `{${filter.string}}`);
+  if (filterWithBrackets !== false) {
+    return {
+      value: filterWithBrackets,
+      string: queryParser.stringify(filterWithBrackets),
+      valid: true,
+    };
+  }
+
+  return filter;
+}
+
+export function isQueryValid(state: QueryBarState) {
+  return QUERY_PROPERTIES.every((prop) => {
+    // When the filter is invalid but can be automatically altered to be valid,
+    // we want to allow the user to run the query.
+    if (prop === 'filter' && state.fields[prop].valid === false) {
+      return (
+        validateField(
```

Comment:
So this will run validate field on a result of validate field that fallbacks to another validate field if the field failed first validation 🤔 maybe we can refactor this a bit and introduce just one `validateFilterField` instead of trying to do this with generic method for a very specific case?